### PR TITLE
Add a different authentication mechanism for S3 bucket

### PR DESCRIPTION
**Pull request by lotoussa**
https://github.com/cloudskiff/driftctl/pull/671

Is this the good place in the doc to add this content ?

### CODE REVIEW

File: docs/usage/cmd/scan.mdx [hunk 1]
```diff
@@ -44,6 +44,16 @@ $ driftctl scan \
   --from tfstate+s3://statebucketdriftctl/terraform.tfstate \
   --from tfstate://terraform_toto.tfstate
 
+# I want to use a dedicated authentication for S3
+# so that driftctl can access Terraform states stored on a different AWS account than the one to scan
+#
+# N.B. Every environment variables to configure the AWS CLI can be override with the DCTL_S3_ prefix.
+# Export a dedicated AWS profile (or any other AWS env var) for accessing S3 as read-only
+$ export DCTL_S3_AWS_PROFILE="<aws_profile_name>"
```
Comment (sundowndev) [inline docs/usage/cmd/scan.mdx:52]: Isn't it `DCTL_S3_PROFILE` instead ?

**Review by sundowndev — CHANGES_REQUESTED** [review]
May be we could add an example for the region as well

File: docs/usage/cmd/scan.mdx (hunk 1 above)
Comment (lotoussa) [reply docs/usage/cmd/scan.mdx:52]: sure, i change it

**Review by sundowndev — APPROVED** [review]

File: docs/usage/cmd/scan.mdx [hunk 2]
```diff
@@ -54,6 +54,9 @@ $ export DCTL_S3_PROFILE="<aws_profile_name>"
 $ export AWS_PROFILE="driftctl-ro"
 $ driftctl scan --from tfstate+s3://mybucket/terraform.tfstate
 
+# You can also change the region to s3 authentication and some more
```
Comment (sundowndev) [inline docs/usage/cmd/scan.mdx:57]: I think this sentence make more sense `You can also use a specific region to authenticate to the S3 bucket`

**Review by sundowndev — CHANGES_REQUESTED** [review]

**Review by sundowndev — APPROVED** [review]

**Review by wbeuil — CHANGES_REQUESTED** [review]
I would have put it under `/providers/aws/authentication` right before the `Terraform custom role` section with this explanation:

# Custom credentials to read a state on an S3 backend

If you want to use a different set of AWS credentials to read your state on S3, you can override each specific AWS environment variable with the DCTL_S3_ prefix. The purpose here is to have the choice to read a state in a different region than your infrastructure. Please don't forget to use your usual AWS credentials to read the resources of your actual infrastructure.

```
# Export a dedicated AWS named profile (or any other AWS environment variables) to read your state in your S3 backend
$ export DCTL_S3_PROFILE="s3reader"
# Export the usual driftctl AWS named profile
$ export AWS_PROFILE="driftctlrole"
$ driftctl scan --from tfstate+s3://mybucket/terraform.tfstate

# You can also use a specific region to authenticate to the S3 bucket
$ DCTL_S3_REGION=us-east-1 driftctl scan --from tfstate+s3://mybucket/terraform.tfstate
```

**Comment by lotoussa** [comment]
> I would have put it under `/providers/aws/authentication` right before the `Terraform custom role` section with this explanation:
> 
> # Custom credentials to read a state on an S3 backend
> If you want to use a different set of AWS credentials to read your state on S3, you can override each specific AWS environment variable with the DCTL_S3_ prefix. The purpose here is to have the choice to read a state in a different region than your infrastructure. Please don't forget to use your usual AWS credentials to read the resources of your actual infrastructure.
> 
> ```
> # Export a dedicated AWS named profile (or any other AWS environment variables) to read your state in your S3 backend
> $ export DCTL_S3_PROFILE="s3reader"
> # Export the usual driftctl AWS named profile
> $ export AWS_PROFILE="driftctlrole"
> $ driftctl scan --from tfstate+s3://mybucket/terraform.tfstate
> 
> # You can also use a specific region to authenticate to the S3 bucket
> $ DCTL_S3_REGION=us-east-1 driftctl scan --from tfstate+s3://mybucket/terraform.tfstate
> ```

I agree, that was my first thought, i'm on it.

**Review by wbeuil — APPROVED** [review]

**Pull request merged**